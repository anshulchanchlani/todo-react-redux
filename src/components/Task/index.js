import React, { Component,Fragment } from 'react';
import './index.scss'
class Task extends Component {
    constructor(props){
        super(props)
        this.handleClose = this.handleClose.bind(this)
    }
    handleClose(){
        this.props.deleteTask(this.props.item.id)
    }
    render() {
        const { item, statusOfTask } = this.props;
        let datetime  = new Date(item.lastModifiedAt);
        let refinedDate = datetime.getDate() + '-' + datetime.getMonth()+1 +'-'+datetime.getFullYear()+' ' + datetime.getHours()+':'+datetime.getMinutes();
        return (
            <Fragment>
            <li className={statusOfTask}>
                <input type="checkbox"
                    name="checkbox"
                    aria-label={item.name}
                    value={item.name}
                    checked={item.completed}
                    onChange={(e) => this.props.handleOnCheck(item.id)}
                    value={item.name} /><span>{item.name}</span>
                   
                    <span aria-label="Close Task"onClick={this.handleClose} className="close-task">&times;</span>
                    <label name="modified date" className="modified-date"> at - {refinedDate}</label>
                     </li>
                     
                     </Fragment>
        );
    }
}

export default Task;