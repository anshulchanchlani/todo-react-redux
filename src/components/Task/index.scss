.close-task{
    float:right;
    cursor:pointer;
    margin-top:-6px;
    font-size:30px;

}

.close-task:hover{
    color:red
}
.modified-date{
    float:right;
    margin-left:20px;
    margin-top:25px;
    font-size:12px;
    color:rgb(119, 116, 116);
   
}