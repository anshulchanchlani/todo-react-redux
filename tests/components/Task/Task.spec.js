import React from 'react';
import Task from '../../../src/components/Task';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {shallow} from 'enzyme';

const item = {
    "id": 1,
    "name": "hi",
    "createdAt": 1547557062930,
    "completed": true,
    "lastModifiedAt": 1547557062930
}
describe('Task tests',()=>{
    const wrapper = shallow(<Task item={item}/>); 
    test('Task renders',()=>{
      expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('Task elements',()=>{
        expect(wrapper.find('input').length).toEqual(1);
    })

   
})
