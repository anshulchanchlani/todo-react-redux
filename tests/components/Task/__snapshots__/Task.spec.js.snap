// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task tests Task renders 1`] = `
<Fragment>
  <li>
    <input
      aria-label="hi"
      checked={true}
      name="checkbox"
      onChange={[Function]}
      type="checkbox"
      value="hi"
    />
    <span>
      hi
    </span>
    <span
      aria-label="Close Task"
      className="close-task"
      onClick={[Function]}
    >
      Ã—
    </span>
    <label
      className="modified-date"
      name="modified date"
    >
       at - 
      15-01-2019 18:27
    </label>
  </li>
</Fragment>
`;
